src/azure-cli/azure/cli/command_modules/acs/_graph.py

```diff
@@ -3,187 +3,39 @@
-    reply_urls=None,
-    key_value=None,
-    key_type=None,
-    key_usage=None,
-    start_date=None,
-    end_date=None,
-    required_resource_accesses=None,
-):
-    password_creds, key_creds = _build_application_creds(password, key_value, key_type, key_usage, start_date, end_date)
-
-    app_create_param = ApplicationCreateParameters(
-        available_to_other_tenants=available_to_other_tenants,
-        display_name=display_name,
-        identifier_uris=identifier_uris,
-        homepage=homepage,
-        reply_urls=reply_urls,
-        key_credentials=key_creds,
-        password_credentials=password_creds,
-        required_resource_access=required_resource_accesses,
-    )
-    try:
-        result = client.create(app_create_param, raw=True)
-        return result.output, result.response.headers["ocp-aad-session-key"]
-    except GraphErrorException as ex:
-        if "insufficient privileges" in str(ex).lower():
-            link = "https://docs.microsoft.com/azure/azure-resource-manager/resource-group-create-service-principal-portal"  # pylint: disable=line-too-long
-            raise AzCLIError(
-                "Directory permission is needed for the current user to register the application. "
-                "For how to configure, please refer '{}'. Original error: {}".format(link, ex)
-            )
-        raise
-
-
-def build_service_principal(rbac_client, cli_ctx, name, url, client_secret):
-    # use get_progress_controller
-    hook = cli_ctx.get_progress_controller(True)
-    hook.add(messsage="Creating service principal", value=0, total_val=1.0)
-    logger.info("Creating service principal")
-    # always create application with 5 years expiration
-    start_date = datetime.datetime.utcnow()
-    end_date = start_date + relativedelta(years=5)
-    result, aad_session_key = create_application(
-        rbac_client.applications, name, url, [url], password=client_secret, start_date=start_date, end_date=end_date
-    )
-    service_principal = result.app_id  # pylint: disable=no-member
-    for x in range(0, 10):
-        hook.add(message="Creating service principal", value=0.1 * x, total_val=1.0)
-        try:
-            create_service_principal(cli_ctx, service_principal, rbac_client=rbac_client)
-            break
-        # TODO figure out what exception AAD throws here sometimes.
-        except Exception as ex:  # pylint: disable=broad-except
-            logger.error(str(ex))
-        time.sleep(2 + 2 * x)
-    else:
-        return False, aad_session_key
-    hook.add(message="Finished service principal creation", value=1.0, total_val=1.0)
-    logger.info("Finished service principal creation")
-    return service_principal, aad_session_key
-
-
-def _create_client_secret():
-    # Add a special character to satisfy AAD SP secret requirements
-    special_char = "$"
-    client_secret = binascii.b2a_hex(os.urandom(10)).decode("utf-8") + special_char
-    return client_secret
+    return result[0]["id"]
```

Comment:
will it possible that the result is not none, but length is zero.

Reply:
No, if this is an empty list, the if statement (empty list is converted to False) at the two lines above will raise an exception.